src/comlink.ts

```diff
@@ -593,25 +595,33 @@ function fromWireValue(value: WireValue): any {
   }
 }
 
+const messageResolvers: Map<string, (value: WireValue | PromiseLike<WireValue>) => void> = new Map();
+
+function handleMessage(ev: Event) {
+    const { data } = ev as MessageEvent;
+    if (!data || !data.id) {
+      return;
+    }
+    const resolver = messageResolvers.get(data.id);
+    if (resolver) {
+        resolver(data);
```

Comment:
I think this would benefit from a try/finally to ensure that you always delete the processed ID.